### PR TITLE
Add check branching validation

### PR DESCRIPTION
**Pull request by maekind**
- .github: repository files added. Release action created to use my new github-action for publishing using `uv`.
- Cli: I have separated the commands and added the option --detect-branches in the validate.
- Validator: I have created the _branches method which will verify if there are branches in the graph.
- Logger: Fix some issues with a verbose option in the CLI.
- CustomFormatter: I have automated the fmt build using its own field definition. Before, fields were duplicated in the custom formatted and logger.
- Tests: Create some more tests to cover new functionality.
- README: updated.

### CODE REVIEW

File: src/pylembic/cli.py [hunk 1]
```diff
@@ -1,45 +1,67 @@
...
 )
 
 
+@app.callback()
+def main(ctx: typer.Context):
+    """
+    Entry point for the CLI tool.
+    """
+    if not ctx.invoked_subcommand:
+        typer.echo("No command specified. Use --help for more information.")
```
Comment (GerardSoleCa) [inline src/pylembic/cli.py]: Not sure if with typer you can do this. Some CLI tools already show the help when no argument or option is used.

File: src/pylembic/cli.py [hunk 2]
```diff
@@ -1,45 +1,67 @@
...
-        False, "--show-graph", help="Visualize the migration dependency graph."
+def validate(
+    migrations_path: str = typer.Argument(
+        default="migrations", help="Path to the migrations folder."
     ),
     verbose: bool = typer.Option(
         False, "--verbose", help="Show migrations validation logs."
     ),
+    detect_branches: bool = typer.Option(
+        False, "--detect-branches", help="Include branching in the validation."
```
Comment (GerardSoleCa) [inline src/pylembic/cli.py]: I'd suggest to improve a bit the help. You are stating that the includes the branching, but not that fails when branches are found

File: src/pylembic/formatter.py [hunk 3]
```diff
@@ -1,15 +1,57 @@
...
+
+        Args:
+            params (list): List of additional parameters to include in the format.
+            base_fmt (str): The base log message format.
+
+        Returns:
+            str: The formatted log message
+        """
+        base_fmt = base_fmt or "%(levelname)s\t %(asctime)s | %(message)s | "
+        dynamic_fields = "".join([f"%({field})s" for field in fields])
```
Comment (GerardSoleCa) [inline src/pylembic/formatter.py:48]: Wouldn't be better to join with a blank space?

```python
dynamic_fields = " ".join([f"%({field})s" for field in fields])
```

Or maybe this is already doing this, just asking/guessing...

File: src/pylembic/validator.py [hunk 4]
```diff
@@ -104,30 +103,67 @@ def _multiple_bases_or_heads(self) -> bool:
...
+                branches = True
+                logger.error(
+                    "Branching migration detected.",
+                    extra={
+                        "migration": node,
+                        "up_revisions": list(self.graph.predecessors(node)),
+                    },
+                )
+
+        return branches
```
Comment (GerardSoleCa) [inline src/pylembic/validator.py:146]: Is it worthy to stop iterating whenever one single branch is found?

File: src/pylembic/formatter.py (hunk 3 above)
Comment (maekind) [reply src/pylembic/formatter.py:48]: In one logging line there will only be one of the fields. So, it’s better to not have spaces in between so we will have weird formatted lines.

File: src/pylembic/validator.py (hunk 4 above)
Comment (maekind) [reply src/pylembic/validator.py:146]: I was going to do that. But then I realised that if I put the verbose option, I would like to see all the branches detected and not only the first one.

File: src/pylembic/cli.py (hunk 1 above)
Comment (maekind) [reply src/pylembic/cli.py]: I didn't like the original message so much, but I guess it's enough and gives more information. So, I reverted this change.

File: src/pylembic/cli.py (hunk 2 above)
Comment (maekind) [reply src/pylembic/cli.py]: I have enriched the message explaining that validation will fail if branches are found.

File: src/pylembic/formatter.py (hunk 3 above)
Comment (GerardSoleCa) [reply src/pylembic/formatter.py:48]: ok, so dynamic_fields will contain only one field to be serialized in the logs?

**Review by GerardSoleCa — APPROVED** [review]

File: src/pylembic/formatter.py (hunk 3 above)
Comment (GerardSoleCa) [reply src/pylembic/formatter.py:48]: Clarified offline

Comment (maekind) [reply src/pylembic/formatter.py:48]: Exactly.

**Pull request merged**